service/src/tests/mocks/enclave_api_mock.rs

```diff
@@ -79,3 +80,17 @@ impl EnclaveBase for EnclaveBaseMock {
 		Ok([1u8; MR_ENCLAVE_SIZE])
 	}
 }
+
+impl Sidechain for EnclaveMock {
+	fn sync_parentchain<ParentchainBlock: ParentchainBlockTrait>(
+		&self,
+		_blocks: &[sp_runtime::generic::SignedBlock<ParentchainBlock>],
+		_nonce: u32,
+	) -> EnclaveResult<()> {
+		Ok(())
+	}
+
+	fn execute_trusted_calls(&self) -> EnclaveResult<()> {
+		todo!()
+	}
+}
```

Comment:
This is a matter of contention: Ideally, your mocks only mock a single trait in order to keep them as specific but also as simple as possible. Whenever you only need the `Sidechain` trait mocked, you now have to use a mock (`EnclaveMock`) that implements more than you potentially want. And it is used by components that require other traits, which creates an indirect dependency.

If, however, all dependents of this trait always need that combination of traits, then it's necessary again (which I'm guessing is the case here?). If so, then we can leave it as is..

Reply:
Hm. I see. Let's leave as it is now, in light of that the Mock currently only used once and that the Traits themselves probably need a refactoring (parentchain_sync should not be Sidechain specific)

My follow-up question will be offline 😈